jplag.frontend.text/src/main/java/de/jplag/text/ParserAdapter.java

```diff
@@ -32,42 +49,75 @@ public TokenList parse(File directory, String[] files) {
+        this.currentFile = file;
+        this.currentLine = 1; // lines start at 1
+        this.currentLineIndex = 0;
+        Path filePath = directory.toPath().resolve(file);
+        String content = readFile(filePath);
+        if (content == null) {
+            return false;
+        }
+        int lastTokenEnd = 0;
+        CoreDocument coreDocument = pipeline.processToCoreDocument(content);
+        for (CoreLabel token : coreDocument.tokens()) {
+            advanceLineBreaks(content, lastTokenEnd, token.beginPosition());
+            lastTokenEnd = token.endPosition();
+            if (isWord(token)) {
+                addToken(token);
+            }
+        }
+        return true;
+    }
+
     /**
-     * Converts a ANTLR token into a JPlag token and adds it to the token list.
-     * @param token is the ANTLR token to convert.
+     * Scan for line breaks and increase {@link #currentLine} and {@link #currentLineIndex} accordingly.
+     * @param content the file content
+     * @param lastTokenEnd the end position of the last token
+     * @param nextTokenBegin the begin position of the next token
      */
-    public void add(Token token) {
-        if (token instanceof AntlrParserToken parserToken) {
-            String text = token.getText();
-            int type = getTokenType(text);
-            tokens.addToken(new TextToken(text, type, currentFile, parserToken));
-        } else {
-            throw new IllegalArgumentException("Illegal token implementation: " + token);
+    private void advanceLineBreaks(String content, int lastTokenEnd, int nextTokenBegin) {
+        for (int i = lastTokenEnd; i < nextTokenBegin; i++) {
+            if (content.charAt(i) == '\n') { // LF
+                currentLine++;
+                currentLineIndex = i;
+            } else if (content.charAt(i) == '\r') { // CR
+                if (i + 1 < content.length() && content.charAt(i + 1) == '\n') { // CRLF
+                    i++; // skip following LF
+                }
+                currentLine++;
+                currentLineIndex = i;
+            }
         }
     }
 
-    private boolean parseFile(File directory, String file) {
-        InputState inputState = null;
-        try (FileInputStream inputStream = new FileInputStream(new File(directory, file))) {
-            currentFile = file;
-            // Create a scanner that reads from the input stream passed to us
-            inputState = new InputState(inputStream);
-            TextLexer lexer = new TextLexer(inputState);
-            lexer.setFilename(file);
-            lexer.setTokenObjectClass("de.jplag.text.AntlrParserToken");
+    private boolean isWord(CoreLabel token) {
+        // consider a token as a word if it contains any alphanumeric character
+        String text = token.originalText();
+        for (int i = 0; i < text.length(); i++) {
+            char c = text.charAt(i);
+            if (Character.isAlphabetic(c) || Character.isDigit(c)) {
```

Comment:
You can use a foreach loop with `String.toCharArray()`. Makes it a bit more cleaner.

Reply:
Or you can use a more functional one liner:
`return text.chars().anyMatch(it -> Character.isAlphabetic(it) || Character.isDigit(it))`